firedrake/solving_utils.py

```diff
@@ -141,6 +141,8 @@ class _SNESContext(object):
     :arg pmat_type: Indicates whether the preconditioner (if present) is assembled
         monolithically ('aij'), as a block sparse matrix ('nest') or
         matrix-free (as :class:`~.ImplicitMatrix`, 'matfree').
+    :arg sub_mat_type: Indicates the matrix type for the sparse blocks
```

Comment:
Can we use numpydoc here?

Reply:
How do we resolve circular dependencies if we need to import the type for type hinting?

Reply:
This might require further thought, I updated it to a numpydoc, but I do not plan to deal with the circular imports in this PR.

Reply:
This is quite a mess in Python, but what we need to do is put
```py
from __future__ import annotations
```
at the top of the file.

This will not be needed in Python 3.14+ (https://peps.python.org/pep-0749/)